stm/stm.ml

```diff
@@ -2509,8 +2509,9 @@ let known_state ~doc ?(redefine_qed=false) ~cache id =
               ), true, true
           | `MaybeASync (start, nodes, name, delegate) -> (fun () ->
                 reach ~cache:true start;
-                (* no sections *)
-                if CList.is_empty (Environ.named_context (Global.env ()))
+                if CList.is_empty (Environ.named_context (Global.env ())) (* no sections *)
+                   || PG_compat.get_pstate () |> (* #[using] attribute *)
+                        Option.cata (fun x -> Option.has_some (Declare.Proof.get_used_variables x)) false
```

Comment:
Why is there a case for the attribute but not for when the `Proof using` command or default proof using occurred?
Can we change it so they have more similar code paths?

Reply:
We could, but it is not trivial. The point is that `collect_proof` scans the AST for a `VernacProof...`. For the attribute one should do something similar but I don't even know if the AST I have here includes the attribute or not.

There is already this codepath that is taken when there is no "Proof using" annotation, and that forces the state at the begin of the proof. Once you have that, it is trivial to fetch the (result of the) attribute.